src/lib/utils/html.ts

```
@@ -0,0 +1,76 @@
+    if (!options) {
+      return defaultXSS.process(html);
+    }
+
+    const defaultOptions: IFilterXSSOptions = {
+      whiteList: { ...DEFAULT_WHITE_LIST },
+    };
+
+    if (options.allowedTags) {
+      defaultOptions.whiteList = options.allowedTags.reduce<Record<string, string[]>>(
+        (array, tag) => ({
+          ...array,
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)